lib/bemhtml/index.js

```diff
@@ -114,16 +112,11 @@ BEMHTML.prototype.render = function render(context,
     out += entity.jsClass;
     out += this.buildModsClasses(entity.block, entity.elem, mods);
 
-    var totalMix = mix;
-    if (ctx.mix) {
-      if (totalMix)
-        totalMix = [].concat(totalMix, ctx.mix);
-      else
-        totalMix = ctx.mix;
-    }
+    if (ctx.mix && mix && mix !== ctx.mix)
+      mix = [].concat(mix, ctx.mix);
 
-    if (totalMix) {
-      var m = this.renderMix(entity, totalMix, jsParams, addJSInitClass);
+    if (mix) {
+      var m = this.renderMix(entity, mix, jsParams, addJSInitClass);
```

Comment:
Yes, of course. `this.exec.mix(context)` now returns this.ctx.mix by default. Therefore we don’t need to have all logic in lines 117—123.